appscale/tools/agents/gce_agent.py

```diff
@@ -1148,34 +1142,44 @@ def open_connection(self, parameters):
     Raises:
       AppScaleException if the user wants to abort.
     """
-    is_autoscale = parameters['autoscale_agent']
+    is_autoscale_agent = parameters.get('autoscale_agent', False)
 
-    # Perform OAuth 2.0 authorization.
-    flow = None
-    if self.PARAM_SECRETS in parameters:
-      secrets_location = os.path.expanduser(parameters[self.PARAM_SECRETS])
-      secrets_type = GCEAgent.get_secrets_type(secrets_location)
+    # Determine paths to credential files
+    if is_autoscale_agent:
+      client_secrets_path = self.CLIENT_SECRETS_LOCATION
+      oauth2_storage_path = self.OAUTH2_STORAGE_LOCATION
+    else:
+      client_secrets_path = LocalState.get_client_secrets_location(
+        parameters[self.PARAM_KEYNAME])
+      oauth2_storage_path = LocalState.get_oauth2_storage_location(
+        parameters[self.PARAM_KEYNAME])
+      if not os.path.exists(client_secrets_path):
+        client_secrets_path = parameters.get(self.PARAM_SECRETS, '')
+      if not os.path.exists(oauth2_storage_path):
+        oauth2_storage_path = parameters.get(self.PARAM_STORAGE, '')
+
+    if os.path.exists(client_secrets_path):
+      # Attempt to perform authorization using Service account
+      secrets_type = GCEAgent.get_secrets_type(client_secrets_path)
       if secrets_type == CredentialTypes.SERVICE:
         scopes = [GCPScopes.COMPUTE]
-        credentials = ServiceAccountCredentials\
-          .from_json_keyfile_name(secrets_location, scopes=scopes)
+        credentials = ServiceAccountCredentials.from_json_keyfile_name(
+          client_secrets_path, scopes=scopes)
         return discovery.build('compute', self.API_VERSION), credentials
-      else:
-        flow = oauth2client.client.flow_from_clientsecrets(secrets_location,
-          scope=self.GCE_SCOPE)
 
-    if is_autoscale in ['True', True]:
-      if not os.path.exists(self.OAUTH2_STORAGE_LOCATION):
-        raise AgentConfigurationException('Could not find your signed OAuth2' \
-          'file at {0}'.format(self.OAUTH2_STORAGE_LOCATION))
-      else:
-        storage = oauth2client.file.Storage(self.OAUTH2_STORAGE_LOCATION)
-    else:
-     storage = oauth2client.file.Storage(LocalState.get_oauth2_storage_location(
-       parameters[self.PARAM_KEYNAME]))
+    # Perform authorization using OAuth2 storage
+    storage = oauth2client.file.Storage(oauth2_storage_path)
     credentials = storage.get()
 
-    if credentials is None or (credentials.invalid in ['True', True]):
+    if not credentials or credentials.invalid:
+      # If we couldn't get valid credentials from OAuth2 storage
+      if os.path.exists(client_secrets_path):
```

Comment:
Should this be `not os.path.exists`?

Reply:
Why didn't I check PR comments before testing?)
Thanks, you're absolutely right.